clientv3/integration/black_hole_test.go

```diff
@@ -26,8 +26,58 @@ import (
 	"go.etcd.io/etcd/integration"
 	"go.etcd.io/etcd/pkg/testutil"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
+// TestServerGRPCKeepAliveTimeout tests ServerParameters keepalive ping on server-side that
+// after having pinged for keepalive check, the server waits for a duration of Timeout and
+// if no activity is seen even after that the connection is closed. The keepalive takes few
+// seconds to get in impact so typically the the time it takes to close the connection is
+// somewhat higher than specified Timeout value.
+func TestServerGRPCKeepAliveTimeout(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{
+		Size:                  2,
+		GRPCKeepAliveInterval: time.Second,
+		GRPCKeepAliveTimeout:  time.Second,
+	})
+	defer clus.Terminate(t)
+
+	eps := []string{clus.Members[0].GRPCAddr()}
+
+	ccfg := clientv3.Config{
+		Endpoints: []string{eps[0]},
+	}
+
+	cli, err := clientv3.New(ccfg)
```

Comment:
Do we need cli in this test, giving that NewClusterV3() already created clients?

Reply:
yes, the client is created here. Thanks!

Reply:
Sorry, I was not able to understand. Do you mean we need this for the desired config for the cluster?  My understanding is that the keepAlive parameters were configured on server side.  And clus.clients are configured inside NewClusterV3.

Reply:
Yes, that's what my understanding is - to override defaults used in  NewClusterV3. In this particular case, it's just the endpoint though and it's more readable this way. Hope makes sense but I am open for suggestion. Thanks!

Reply:
Agree, I tried to fix your concern in the new commit. Thanks!